static/js/stream_list.js

```diff
@@ -524,6 +525,15 @@ export function set_event_handlers() {
         .on("click", (e) => {
             e.preventDefault();
             if (e.target.id === "streams_inline_cog") {
+                if (
+                    !(
+                        settings_data.user_can_create_private_streams() ||
+                        settings_data.user_can_create_public_streams() ||
+                        settings_data.user_can_create_web_public_streams()
+                    )
+                ) {
+                    window.location.assign("#streams/all");
+                }
                 return;
             }
             toggle_filter_displayed(e);
```

Comment:
It might help to rework this function using the early return style, i.e. 

```
if (...) { 
    return; 
}
```

just to avoid the readability issues that come with deep nesting of conditionals.

Reply:
@timabbott sorry for responding late, I was having a little difficulty in understanding the assignment. I have added another event handler to `#streams_inline_cog` as per the requirements.